src/main/java/com/github/manolo8/darkbot/gui/titlebar/FlashRunnerTask.java

```diff
@@ -0,0 +1,81 @@
+package com.github.manolo8.darkbot.gui.titlebar;
+
+import com.github.manolo8.darkbot.backpage.BackpageManager;
+import com.github.manolo8.darkbot.utils.LibSetup;
+import com.github.manolo8.darkbot.utils.http.Method;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FlashRunnerTask extends Thread {
+    private static final Path RUNNER_PATH = Paths.get("lib", "KekkaRunner.exe");
+    private static final Pattern PARAMS_PATTERN = Pattern.compile("src\":.\"([^\"]+).*width\":.(\\d+).*height\":.(\\d+).*(cdn[^}]+)");
+
+    private static boolean LIB_CHECKED = false;
+
+    private final BackpageManager backpageManager;
+    private final JMenuItem menuItem;
+    private final String name;
+
+    public FlashRunnerTask(BackpageManager backpageManager, JMenuItem menuItem, String name) {
```

Comment:
Instead of passing a JMenuItem, i would pass a `Consumer<Boolean> callback`, and call with true/false for ok or fail. This unties the logic of the runner to the actuall trigger (the menu item/gui), and allows it to be called from more than just the menu.

Additionally, the default callback should be a `result -> SwingUtilities.invokeLater(() -> menu.setEnable(true))` because we need that enabled state change to happen on the swing thread